periodic/periodic.go

```diff
@@ -219,9 +233,24 @@ func runOne(id int, funcTimes *stats.Histogram, sleepTimes *stats.Histogram, num
 	perThreadQPS := r.QPS / float64(r.NumThreads)
 	useQPS := (perThreadQPS > 0)
 	f := r.Function
+
+	// Catch SIGINT signals from the OS and raise a flag to terminate the run loop
+	terminated := false
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, os.Interrupt)
+	go func() {
+		<-c
+		terminated = true
```

Comment:
more importantly: while this works on x86 I don't think it's safe (unprotected read/write across go routines)

Reply:
@ldemailly Thanks for the -race tip!